File: Week-07/Day-2/student-counter.js
'use strict';

var students = [
        {'name': 'Teodor', 'age': 3, 'candies': 2},
        {'name': 'Rezso', 'age': 9.5, 'candies': 2},
        {'name': 'Zsombor', 'age': 12, 'candies': 5},
        {'name': 'Aurel', 'age': 7, 'candies': 3},
        {'name': 'Olaf', 'age': 12, 'candies': 7},
        {'name': 'Gerzson', 'age': 10, 'candies': 1},
]

// create a function that takes a list of students and logs: 
// - how many candies are owned by students
function candyCounter(st_list) {
    let candies = 0;
    for (let i = 0; i < st_list.length; i += 1) {
        candies += st_list[i]['candies'];
    }
    return candies;
}
console.log(candyCounter(students))
// create a function that takes a list of students and logs:
// - Sum of the age of people who have lass than 5 candies
function sum_age(st_list) {
    let age = 0;
    for (let i = 0; i < st_list.length; i++) {
        if (st_list[i]['candies'] < 5) {
            age += st_list[i]['age'];
        }
    }
    return age;
}

console.log(sum_age(students));

File: Week-02/Day-3/student_counter.py
students = [
        {'name': 'Teodor', 'age': 3, 'candies': 2},
        {'name': 'Rezso', 'age': 9.5, 'candies': 2},
        {'name': 'Zsombor', 'age': 12, 'candies': 5},
        {'name': 'Aurel', 'age': 7, 'candies': 3},
        {'name': 'Olaf', 'age': 12, 'candies': 7},
        {'name': 'Gerzson', 'age': 10, 'candies': 1},
]

# create a function that takes a list of students and prints: 
# - how many candies are owned by students

# create a function that takes a list of students and prints:
def candies():
    for i in range(len(students)):
        print (students[i]['name'] + ' : ' + str(students[i]['candies']))

candies ()
# create a function that takes a list of students and prints:
# - Sum of the age of people who have less than 5 candies

def age_sum():
    total = 0
    for i in range(len(students)):
        if students[i]['candies'] < 5:
            total += students[i]['age']
    print(total)
age_sum()